code/pilotfile/pilotfile.cpp

```diff
@@ -281,7 +281,7 @@ void pilotfile::update_stats_backout(scoring_struct *stats, bool training)
 
 			for (idx = 0; idx < list_size; idx++) {
 				if ( p_stats->ship_kills[idx].name.compare(Ship_info[i].name) == 0 ) {
-					j = i;
+					j = idx;
```

Comment:
`j` is used as an index for `p_stats->ship_kills` a few lines down. `i` is an index into `Ship_info`. As far as I can tell `p_stats->ship_kills` and `Ship_info` are not necessarily the same size so using `i` as an index into `ship_kills` doesn't seem like the intended function.